base/MAPL_Resource.F90

```diff
@@ -0,0 +1,411 @@
+               val = trim(default)
+            end select
+         else
+            call ESMF_ConfigGetAttribute(config, val, label = label, _RC)
+         end if
+      type is(logical)
+         if (default_is_present .and. .not. label_is_present) then
+            select type(default)
+            type is(logical)
+               val = default
+            end select
+         else
+            call ESMF_ConfigGetAttribute(config, val, label = label, _RC)
+         end if
+         class default
+         _FAIL( "Unupported type")
+      end select
+
+      call ESMF_ConfigGetAttribute(config, printrc, label = 'PRINTRC:', default = 0, _RC)
+
+      ! Can set printrc to negative to not print at all
+      if (MAPL_AM_I_Root() .and. printrc >= 0) then
+         if (label_is_present) then
+            label_to_print = label
+         else
+            label_to_print = trim(label_to_find)
+         end if
+         call print_resource(printrc, label_to_print, val, default=default, _RC)
+      end if
+
+      _RETURN(ESMF_SUCCESS)
+
+   end subroutine MAPL_GetResource_config_scalar
+
+   ! Find value of array variable in config
+   subroutine MAPL_GetResource_config_array(config, compname, vals, label_to_find, default, rc)
+      type(ESMF_Config), intent(inout) :: config
+      character(len=*), intent(in) :: compname
+      character(len=*), intent(in) :: label_to_find
+      class(*), intent(inout) :: vals(:)
+      class(*), optional, intent(in) :: default(:)
+      integer, optional, intent(out) :: rc
+
+      character(len=:), allocatable :: label_to_use
+      integer :: status, count
+      logical :: label_is_present, default_is_present
+
+      default_is_present = present(default)
+
+      if (default_is_present) then
+         _ASSERT(same_type_as(vals, default), "Value and default must have same type")
+      end if
+
+      call get_label_to_use(config, label_to_find, compname, label_is_present, label_to_use, _RC)
+
+      ! No default and not in config, error
+      ! label or default must be present
+      if (.not. label_is_present .and. .not. default_is_present) then
+         if (present(rc)) rc = ESMF_FAILURE
+         return
+      end if
+
+      count = size(vals)
+
+      select type(vals)
+      type is(integer(int32))
+         if (default_is_present .and. .not. label_is_present) then
+            select type(default)
+            type is(integer(int32))
+               if (.not. label_is_present) vals = default
+            end select
+         else
+            call ESMF_ConfigGetAttribute(config, valuelist = vals, count = count, label = label_to_use, _RC)
+         end if
+      type is(integer(int64))
+         if (default_is_present .and. .not. label_is_present) then
+            select type(default)
+            type is(integer(int64))
+               vals = default
+            end select
+         else
+            call ESMF_ConfigGetAttribute(config, valuelist = vals, count = count, label = label_to_use, _RC)
```

Comment:
Same comments on this massive block as for the scalar case.

Reply:
Done. Separate macro because of slight difference.